domain/service/json.go

```diff
@@ -58,21 +58,24 @@ func (e *ServiceEndpoint) UnmarshalJSON(b []byte) error {
 	} else {
 		return err
 	}
-	glog.V(4).Infof("ServiceEndpoint UnmarshalJSON %#v", e)
+	logger := plog.WithField("endpointname", e.Name)
 	if len(e.VHostList) > 0 {
 		//VHostList is defined, keep it and unset deprecated field if set
 		e.VHosts = nil
 		return nil
 	}
 	if len(e.VHosts) > 0 {
 		// no VHostsList but vhosts is defined. Convert to VHostsList
-		if glog.V(2) {
-			glog.Warningf("ServiceEndpoint VHosts field is deprecated, see VHostList: %#v", e.VHosts)
+		if log.GetLevel() == log.DebugLevel {
+			logger.WithField("vhosts", e.VHosts).Warning("The field named VHosts in ServiceEndpoint is deprecated, see VHostList")
```

Comment:
I think the idea is that in the normal case (no debug), we will silently migrate the data for you if you have an older service definition, but if you turn on debug, it's a warning to indicate that while we upgraded the data, we really don't expect to encounter very many of these cases in practice (i.e. you have a pretty old service definition if we're migrating this stuff on the fly).